R/helper-functions.R

```diff
@@ -485,35 +548,78 @@ create_sparklines <- function(
       xml_node_create("x14:colorNegative", xml_attributes = color_negative),
-      xml_node_create("x14:colorAxis",     xml_attributes = color_axis),
-      xml_node_create("x14:colorMarkers",  xml_attributes = color_markers),
-      xml_node_create("x14:colorFirst",    xml_attributes = color_first),
-      xml_node_create("x14:colorLast",     xml_attributes = color_last),
-      xml_node_create("x14:colorHigh",     xml_attributes = color_high),
-      xml_node_create("x14:colorLow",      xml_attributes = color_low),
+      xml_node_create("x14:colorAxis", xml_attributes = color_axis),
+      xml_node_create("x14:colorMarkers", xml_attributes = color_markers),
+      xml_node_create("x14:colorFirst", xml_attributes = color_first),
+      xml_node_create("x14:colorLast", xml_attributes = color_last),
+      xml_node_create("x14:colorHigh", xml_attributes = color_high),
+      xml_node_create("x14:colorLow", xml_attributes = color_low),
       xml_node_create(
-        "x14:sparklines", xml_children = c(
-          xml_node_create(
-            "x14:sparkline", xml_children = c(
-              xml_node_create(
-                "xm:f", xml_children = c(
-                  paste0(shQuote(sheet, type = "sh"), "!", dims)
-                )),
-              xml_node_create(
-                "xm:sqref", xml_children = c(
-                  sqref
-                ))
-            ))
-        )
+        "x14:sparklines",
+        xml_children = sparklines
       )
     )
   )
 
   sparklineGroup
 }
 
+# Helper to split a cell range into rows or columns
+split_dims <- function(x, direction = "row") {
+  match.arg_wrapper(
+    direction,
+    c("row", "col"),
+    fn_name = "dims_split",
+    arg_name = "direction"
+  )
+
+  rc <- dims_to_rowcol(x, as_integer = FALSE)
+
+  cols <- col2int(rc[[1]])
+  rows <- as.integer(rc[[2]])
+
+  if (identical(length(rows), 1L) || identical(length(cols), 1L)) {
+    return(rowcol_to_dims(row = rows, col = cols))
+  }
+
+  if (identical(direction, "row")) {
+    vapply(rows, function(row) {
+      rowcol_to_dims(row = row, col = cols)
+    }, FUN.VALUE = character(1))
+  } else {
+    vapply(cols, function(col) {
+      rowcol_to_dims(row = rows, col = col)
+    }, FUN.VALUE = character(1))
+  }
+}
+
+# Helper to split a cell range of one row or column into single cells
+split_dim <- function(x) {
+  rc <- dims_to_rowcol(x, as_integer = FALSE)
+
+  cols <- col2int(rc[[1]])
+  rows <- as.integer(rc[[2]])
+
+  if (length(cols) > 1 && length(rows) > 1) {
+    stop("`x` should be a cell range of one row or one column.")
+  }
+
+  col_min <- min(cols)
+  row_min <- min(rows)
+
+  if (identical(length(cols), 1L)) {
+    vapply(rows, function(row) {
+      rowcol_to_dim(row = row, col = col_min)
+    }, FUN.VALUE = character(1))
+  } else {
+    vapply(cols, function(col) {
+      rowcol_to_dim(row = row_min, col = col)
+    }, FUN.VALUE = character(1))
+  }
+}
+
```

Comment:
My approach to this would be something like this.  Avoids the conversion to integers:
``` r
split_dims <- function(dims, direction = c("row", "col")) {
  if (is.numeric(direction)) {
    if (direction == 1) direction <- "row"
    if (direction == 2) direction <- "col"
  }
  df <- openxlsx2:::dims_to_dataframe(dims, fill = TRUE)
  direction <- match.arg(direction)
  if (direction == "row") df <- as.data.frame(t(df))
  vapply(df, FUN = function(x) {
    fst <- x[1]
    snd <- x[length(x)]
    sprintf("%s:%s", fst, snd)
  }, FUN.VALUE = NA_character_)
}

split_dim <- function(dims) {
  df <- openxlsx2:::dims_to_dataframe(dims, fill = TRUE)
  if (ncol(df) > 1 && nrow(df) > 1) 
    stop("`dims` should be a cell range of one row or one column.", call. = FALSE)
  df
}

dims <- "A1:C3"

split_dims(dims, 1)
#>       1       2       3 
#> "A1:C1" "A2:C2" "A3:C3"
split_dims(dims, 2)
#>       A       B       C 
#> "A1:A3" "B1:B3" "C1:C3"
split_dim(dims)
#> Error: `dims` should be a cell range of one row or one column.

dims <- "A1:A3"

split_dims(dims, "row")
#>       1       2       3 
#> "A1:A1" "A2:A2" "A3:A3"
split_dims(dims, "col")
#>       A 
#> "A1:A3"
split_dim(dims)
#>    A
#> 1 A1
#> 2 A2
#> 3 A3

dims <- "A1:C1"
split_dims(dims, "row")
#>       1 
#> "A1:C1"
split_dims(dims, "col")
#>       A       B       C 
#> "A1:A1" "B1:B1" "C1:C1"
split_dim(dims)
#>    A  B  C
#> 1 A1 B1 C1
```

Reply:
No objections. I will adopt your suggestion.